pkgs/development/python-modules/hatchling/default.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pythonOlder
+, isPy27
+
+# All runtime dependencies
+, editables
+, importlib-metadata
+, packaging
+, pathspec
+, pluggy
+, toml
+, tomli
+}:
+
+buildPythonPackage rec {
+  pname = "hatchling";
+  version = "0.19.0";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "942b34e454ce3be5b03139e0a198ba8ebc08a6095d345fe829b1ae92e9326c2c";
+  };
+
+  propagatedBuildInputs = [
+    packaging
+    pathspec
+    pluggy
+  ] ++ lib.optionals (pythonOlder "3.8") [
+    importlib-metadata
+  ] ++ lib.optionals isPy27 [
```

Comment:
forgot to put this into my previous review but we would like to remove/minimize python2 packages from nixpkgs so please no new python2 packages
https://github.com/NixOS/nixpkgs/projects/27